addons/mail/static/src/model/model_manager.js

```diff
@@ -283,23 +254,32 @@ class ModelManager {
      *
      * @param {mail.model} Model class
      * @param {string} localId
+     * @param {Object} param2
+     * @param {boolean} [param2.isCheckingInheritance=false]
      * @returns {mail.model|undefined} record, if exists
      */
-    get(Model, localId) {
+    get(Model, localId, { isCheckingInheritance = false } = {}) {
```

Comment:
I think we should get rid of the support of inheritance for relational fields, and consider it deprecated until dialog is refactored. It will only introduce more troubles, both conceptually and with performance.

Reply:
I'm in favour of not using inherited models, but we will likely need to define categories among records. Like with dialog, could imagine a relation to any UI record. Could even imagine relational field could be on a class or a category (or only category, and classes have auto-generated category named like the model name itself).

Global store or records in manager + "indexes" in each Model would go in that direction.